### PR TITLE
CSEN-20 Request classes for User was implemented

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+*/
+package com.bwsw.cloudstack.entities.requests
+
+import java.util.{TimeZone, UUID}
+
+import br.com.autonomiccs.apacheCloudStack.client.ApacheCloudStackRequest
+import com.bwsw.cloudstack.entities.requests.traits.Request
+
+/**
+  * Class is responsible to create ApacheCloudStackRequest with specified parameters for creating user
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: Class is responsible for building ApacheCloudStackRequest

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  *
+  * @param settings mandatory parameters for user creation, more info see UserCreateRequest.Settings
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: mandatory -> required

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 1 above)
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: responsible for (not to) - change it everywhere

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  */
+class UserCreateRequest(settings: UserCreateRequest.Settings) extends Request {
+  override val request = new ApacheCloudStackRequest("createUser")
+    .addParameter("response","json")//toZoneId
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: remove the comment

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    .addParameter("account", settings.accountName)
+    .addParameter("email", settings.email)
+    .addParameter("firstname", settings.firstName)
+    .addParameter("lastname", settings.lastName)
+    .addParameter("password", settings.password)
+    .addParameter("username", settings.username)
+
+  /**
+    * Add domain id parameter into request.
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: to*

Comment (placccebo) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: a request

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    request.addParameter("userid", id)
+    this
+  }
+}
+
+object UserCreateRequest {
+  /**
+    * Class is responsible to provide user creation settings
+    *
+    * @param accountName provides to create the user under the specified account.
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: user account name. If it isn't specified, a username will be used as an account name.

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 3 above)
Comment (placccebo) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: use a class to store parameter names as constants

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserFindRequest.scala [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    * Add user id parameter into request
+    */
+  def withId(id: UUID): UserFindRequest = {
+    request.addParameter("id", id)
+    this
+  }
+
+  /**
+    * Add account name parameter into request,
+    * must be used with domain id parameter
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/requests/UserFindRequest.scala]: remove this line

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    new ApacheCloudStackApiCommandParameter("response","json"),
+    new ApacheCloudStackApiCommandParameter("account", settings.accountName),
+    new ApacheCloudStackApiCommandParameter("email", settings.email),
+    new ApacheCloudStackApiCommandParameter("firstname", settings.firstName),
+    new ApacheCloudStackApiCommandParameter("lastname", settings.lastName),
+    new ApacheCloudStackApiCommandParameter("password", settings.password),
+    new ApacheCloudStackApiCommandParameter("username", settings.username)
+  )
+
+  "Instance creation" should "create request with default parameters" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: it should create a request with predefined and specified (via constructor) parameters

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+import scala.collection.JavaConverters._
+
+class UserFindRequestTestSuite extends FlatSpec {
+  val defaultParameters = Set[ApacheCloudStackApiCommandParameter](
+    new ApacheCloudStackApiCommandParameter("response","json"),
+    new ApacheCloudStackApiCommandParameter("listAll","true")
+  )
+
+  "Instance creation" should "create request with 'response' and 'listAll' default parameters" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala]: see the comment above

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserFindRequest.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserFindRequest.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala (hunk 7 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala (hunk 8 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequest.scala]: ok

**Comment by MedvedevBW** [comment]
@placccebo

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+  "withId" should "add id parameter to a request" in {
+    val userId = UUID.randomUUID()
+    val expectedParameters = defaultParameters ++ Set(new ApacheCloudStackApiCommandParameter(USER_ID, userId.toString))
+    val request = new UserCreateRequest(settings)
+
+    assert(request.withId(userId).request.getParameters.asScala.toSet == expectedParameters)
+  }
+
+  "withAccountName" should "add account name parameter to a request" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: fix it

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    val timezone = "GMT+0700"
+    val expectedParameters = defaultParameters ++ Set(new ApacheCloudStackApiCommandParameter(TIMEZONE, timezone))
+    val request = new UserCreateRequest(settings)
+
+    assert(request.withTimeZone(TimeZone.getTimeZone(timezone)).request.getParameters.asScala.toSet == expectedParameters)
+  }
+
+  "withDomain" should "add account name parameter to a request" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: fix it

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+  "withAccountName" should "add account name parameter to a request" in {
+    val accountName = "test"
+    val expectedParameters = defaultParameters ++ Set(new ApacheCloudStackApiCommandParameter(ACCOUNT, accountName))
+    val request = new UserFindRequest
+
+    assert(request.withAccountName(accountName).request.getParameters.asScala.toSet == expectedParameters)
+  }
+
+  "withDomain" should "add account name parameter to a request" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala]: fix it

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala (hunk 9 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala (hunk 10 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/UserCreateRequestTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala (hunk 11 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/requests/UserFindRequestTestSuite.scala]: ok

**Comment by MedvedevBW** [comment]
@placccebo

**Review by placccebo — APPROVED** [review]

**Pull request merged**